src/components/container/RelativeActivity/RelativeActivity.tsx

```diff
@@ -0,0 +1,111 @@
+import { faChevronUp } from "@fortawesome/free-solid-svg-icons";
+import { startOfDay } from "date-fns";
+import React, { useContext } from "react";
+import { useState } from "react";
+import { FontAwesomeSvgIcon } from "react-fontawesome-svg-icon";
+import getDayKey from "../../../helpers/get-day-key";
+import { getDailyDataTypeDefinition } from "../../../helpers/query-daily-data";
+import { ActivityMeter, CardTitle, DateRangeContext, } from "../../presentational";
+import "./RelativeActivity.css"
+import language from "../../../helpers/language";
+import { useInitializeView } from "../../../helpers/Initialization";
+import { RelativeActivityContext } from "../RelativeActivityDayCoordinator/RelativeActivityDayCoordinator";
+import { RelativeActivityDataType, RelativeActivityQueryResult } from "../../../helpers";
+import { queryRelativeActivity } from "../../../helpers/relative-activity";
+
+export interface RelativeActivityProps {
+    dataTypes?: RelativeActivityDataType[];
+    useContext?: boolean;
+    previewState?: "Default";
+    title?: string;
+    innerRef?: React.Ref<HTMLDivElement>
+    date?: Date;
+}
+
+export default function (props: RelativeActivityProps) {
+    let [results, setResults] = useState<{ [key: string]: RelativeActivityQueryResult } | undefined>(undefined);
+    let relativeActivityContext = useContext(RelativeActivityContext);
+
+    let dataTypes: RelativeActivityDataType[] = props.dataTypes || [];
+    if (props.useContext && relativeActivityContext?.dataTypes) {
+        dataTypes = relativeActivityContext.dataTypes;
+    }
+
+    let dateRangeContext = useContext(DateRangeContext);
+    let date = props.date ?? dateRangeContext?.intervalStart ?? startOfDay(new Date());
+
+    function loadData() {
+        function transformResults(results: { [key: string]: { [key: string]: RelativeActivityQueryResult } } | undefined) {
+            if (!results) return;
+            let computedResult: { [key: string]: RelativeActivityQueryResult } = {};
+            dataTypes.forEach(dataType => {
+                computedResult[dataType.dailyDataType] = results[dataType.dailyDataType][getDayKey(date!)];
+            });
+            return computedResult;
+        }
+
+        setResults(undefined);
+        if (props.useContext) {
+            setResults(transformResults(relativeActivityContext!.data));
+            return;
+        }
+        queryRelativeActivity(date!, date!, dataTypes, !!props.previewState).then(results => {
+            setResults(transformResults(results));
+        });
+    }
+
+    useInitializeView(() => {
+        loadData();
+    }, ["externalAccountSyncComplete"], [dateRangeContext, relativeActivityContext, props.date, props.dataTypes], 0);
```

Comment:
I'm curious about why you'd want to disable the debounce here by passing `0`.  The debounce in `useInitializeView` is meant to prevent double loading when an MDH tab first renders and a couple of events fire almost simultaneously.  It seems unlikely that a second event firing within 500ms of the first would result in a query that returns additional data.

Reply:
So yes, this actually is an issue I wanted to surface, because I think it's broader than this and may actually have effects on other places we use this.  You can see it if you actually set this back to 500 and check out the RelativeActivityDayCoordinator > WithRelativeActivity story.

Because we are listening for context changes here, when you click on each day, it actually takes 500ms to update the selected day which definitely is not a good UX - the actual selected day should update immediately.  Ultimately, context changes can result from user interaction so should happen immediately rather than having a debounce.

So yeah, i'm actually not sure what the best workaround is here.  I considered a couple things:
- We could make it a "leading" debounce, but of course if someone clicks a second day within 500ms (possible) that means it won't fire at all.  
- We could make the debounce only happen for the MyDataHelps events, e.g. instead of the useEffect calling debouncedInitialize() we call initialize() - I don't think this prevents a double render on first load if the app also fires applicationDidBecomeVisible though

Maybe other answers like disabling the debounce after the first debounce if this is just to prevent issues with initial load or something?

Reply:
What *might* be the answer is this:

- Determine whether this is the "initial mount" of the component via https://stackoverflow.com/a/55075818
- On the initial mount of the component, call debouncedInitialize in the useEffect
- On subsequent useEffect calls, just call initialize
- MyDataHelps.on stuff always uses debouncedInitialize

I'll see if that works

Reply:
Seems like it does work - see adjustments to useInitializeView - wondering if you have any thoughts on that though since i'm not as familiar with the scenarios for why we added the debounce

Reply:
Yeah, I like that and I think it should work fine.

The initial load and MDH events would all be debounced, and other subsequent re-executions of `useEffect` should only be caused by dependency changes, I think, which should be okay to just use initialize immediately.